### PR TITLE
chore: add GH template boilerplate

### PR DESCRIPTION
**Pull request by LesnyRumcajs**
Changes:
- added some boilerplate to make this repo more of a first-class citizen,
- added markdown formatter to the CI, fixed offenses,
- generic ignore file
- moved some README files to API given they don't give much description and are auto-generated.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,35 +14,42 @@ You may [browse the spec on the OpenRPC playground](https://playground.open-rpc.
 This repo also contains tooling for creating, interacting with,
 and testing against the schema.
 
-- [`cna-tool`](src/tool/README.md)
-- [`cna-test-suite`](src/test-suite/README.md)
-- [`cna-util`](src/test-suite/README.md)
```
Comment (LesnyRumcajs) [inline README.md:19]: this was wrong.

File: src/test-suite/src/main.rs [hunk 2]
```diff
@@ -744,7 +744,7 @@ mod harness {
 
 #[test]
 fn doc() {
-    expect_test::expect_file!["../README.md"].assert_eq(&cna_util::markdown(
+    expect_test::expect_file!["../API.md"].assert_eq(&cna_util::markdown(
```
Comment (LesnyRumcajs) [inline src/test-suite/src/main.rs:747]: Moved to files that would more likely be understood as auto-generated (and it's easier to put it to prettify)

File: .github/ISSUE_TEMPLATE/task.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+---
+name: Task
+about: "[internal] A specific, actionable unit of work"
```
Comment (ansermino) [inline .github/ISSUE_TEMPLATE/task.md:3]:
```suggestion
about: "[Internal] A specific, actionable unit of work"
```

File: .github/ISSUE_TEMPLATE/epic.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
+---
+name: Epic
+about: [Internal] Larger chunk of work that can be broken down into a smaller tasks
```
Comment (ansermino) [inline .github/ISSUE_TEMPLATE/epic.md:3]:
```suggestion
about: "[Internal] Larger chunk of work that can be broken down into smaller tasks"
```

**Review by ansermino — CHANGES_REQUESTED** [review]
Please update issue template naming as per https://github.com/ChainSafe/forest/pull/4846

**Review by ansermino — APPROVED** [review]

**Pull request merged**